system/lib/libc/musl/src/env/__environ.c

```diff
@@ -1,16 +1,17 @@
 #include "libc.h"
 
-#ifdef __EMSCRIPTEN__
-#include <stdlib.h>
-#include <wasi/api.h>
-#endif
-
 char **__environ = 0;
 weak_alias(__environ, ___environ);
 weak_alias(__environ, _environ);
 weak_alias(__environ, environ);
 
 #ifdef __EMSCRIPTEN__
+#include <stdlib.h>
+#include <wasi/api.h>
+#include <emscripten/emmalloc.h>
```

Comment:
why does this include emmalloc?

Reply:
That is where emscripten_builtin_malloc is defined.

Reply:
Oh, really? That's quite odd, as that header is for the `emmalloc` replacement for `dlmalloc`, which we don't even use by default.

It looks like the function is declared also in `dlmalloc.c`, but not in a header. We should probably create a new header just for that method, I think? For now to not block you, how about a comment here with a TODO to fix this?

Reply:
Yes it does seem a little odd.   I think maybe the idea was that `emmalloc.h` contains all the malloc-related functions?    `emmalloc.h` also defines malloc and free as well BTW... i think it made sense maybe to define all 3 variants in the same place?

I didn't originally put them there .. maybe there is a better place?  `emscripten/heap.h`?  For now I think its fine.  Certainly unrelated to this PR.  We already include `emmalloc.h` in pthread _create.c for this same purpose: 

https://github.com/emscripten-core/emscripten/blob/6b53dedf3d2ace15bfd780e44f4a8ba34407fc27/system/lib/pthread/pthread_create.c#L16